backend/src/features/auth/auth.service.ts

```diff
@@ -52,6 +52,11 @@ export class AuthService {
             name: oryInfo.display_name || oryInfo.username || oryInfo.sub.split('@')[0],
             profileImage: oryInfo.avatar,
             willSyncFromMezon: false,
+            mezonUserId: oryInfo.user_id
+          });
+        } else {
```

Comment:
split logic, update mezon id is a seperate clause

```suggestion
        if (user.willSyncFromMezon) {
          await this.userRepository.update(user.id, {
            name: oryInfo.display_name || oryInfo.username || oryInfo.sub.split('@')[0],
            profileImage: oryInfo.avatar,
            willSyncFromMezon: false
          });
        }
        
        if (!user.mezonUserId) {
          await this.userRepository.update(user.id, {
            mezonUserId: oryInfo.user_id
          });
        }
```